src/Conversion/ONNXToKrnl/Tensor/Pad.cpp

```diff
@@ -68,11 +69,22 @@ struct ONNXPadOpLowering : public OpConversionPattern<ONNXPadOp> {
       // This way is to avoid using `select` in computing indices as doing for
       // 'edge' and 'reflect' modes.
       Value cValue;
+      SmallVector<Value, 1> loadIndices;
+      if (MemRefType constantValueType =
+              constantValue.getType().dyn_cast<MemRefType>()) {
+        if (constantValueType.getElementType().isF32() &&
+            constantValueType.getRank() == 1) {
+          // If the constant_value type is 1xf32 do krnl.load with an index of
+          // 0 to avoid an assertion failure in AffineLoadOp::build().
+          Value idx = create.math.constantIndex(0);
+          loadIndices = {idx};
+        }
+      }
```

Comment:
I would move this section of code into the else part of isa<NoneType> because you need this only when the type isa<NoneType> is false.

Reply:
Moved this code into else part of isa.  Thank you.